airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/util/WriteStreamConsumer.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.core.util
+
+import io.airbyte.cdk.core.context.env.ConnectorConfigurationPropertySource
+import io.airbyte.cdk.integrations.base.SerializedAirbyteMessageConsumer
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.micronaut.context.annotation.Requires
+import jakarta.inject.Named
+import jakarta.inject.Singleton
+import java.io.InputStream
+import java.nio.charset.StandardCharsets
+
+private val logger = KotlinLogging.logger {}
+
+@Singleton
+@Named("writeOperationExecutor")
+@Requires(
+    property = ConnectorConfigurationPropertySource.CONNECTOR_OPERATION,
+    value = "write",
+)
+@Requires(env = ["destination"])
+class WriteStreamConsumer(private val consumer: SerializedAirbyteMessageConsumer) {
+
+    fun consumeWriteStream(inputStream: InputStream = System.`in`) {
+        logger.info { "Starting buffered read of input stream" }
+        consumer.start()
+        inputStream.bufferedReader(StandardCharsets.UTF_8).use {
+            var emptyLines = 0
+            it.lines().forEach { line: String ->
+                if (line.isNotEmpty()) {
+                    consumer.accept(line, line.toByteArray(StandardCharsets.UTF_8).size)
+                } else {
+                    emptyLines++
+                    // We've occasionally seen this loop not exit
+                    // maybe it's because we keep getting streams of empty lines?
+                    // TODO: Monitor the logs for occurrences of this log line and if this isn't
+                    // an issue, remove it.
+                    if (emptyLines % 1_000 == 0 && emptyLines < 10_000) {
```

Comment:
super unfair comment that is not blocking - I was thinking about this last night and these conditions should probably be reversed since once its above 10k I didn't want to keep logging